subsys/bluetooth/mesh/blob_io_flash.c

```diff
@@ -56,40 +56,55 @@ static void io_close(const struct bt_mesh_blob_io *io,
 	flash_area_close(flash->area);
 }
 
+static inline int erase_device_block(const struct flash_area *fa, off_t start, size_t size)
+{
+	/* If there are no devices requiring erase, then there is nothing to do */
+	if (IS_ENABLED(CONFIG_BT_MESH_BLOB_IO_FLASH_WITH_ERASE)) {
+		const struct device *fdev = flash_area_get_device(fa);
+
+		if (!fdev) {
+			return -ENODEV;
+		}
+
+		/* We have a mix of devices in system */
```

Comment:
In which `system`? How is it related to ble mesh?

Reply:
System as a set of features/drivers enabled in Zephyr running on a hardware.